### PR TITLE
Mobile support

### PR DESCRIPTION
**Pull request by Fyzu**
Ref #124

### CODE REVIEW

File: src/ObsidianOutlinerPlugin.ts [hunk 1]
```diff
@@ -37,13 +37,13 @@ export default class ObsidianOutlinerPlugin extends Plugin {
 
     this.obsidian = new ObsidianService(this.app);
 
-    if (this.obsidian.isLegacyEditorEnabled()) {
```
Comment (Fyzu) [inline src/ObsidianOutlinerPlugin.ts:40]: Is it ok to remove this part of the code? Or how should we handle it on mobile to allow running the plugin?

File: src/ObsidianOutlinerPluginWithTests.ts [hunk 2]
```diff
@@ -102,10 +102,10 @@ export default class ObsidianOutlinerPluginWithTests extends ObsidianOutlinerPlu
     (window as any).ObsidianOutlinerPlugin = this;
 
     if (process.env.TEST_PLATFORM) {
-      setImmediate(async () => {
+      setTimeout(async () => {
```
Comment (Fyzu) [inline src/ObsidianOutlinerPluginWithTests.ts:105]: `setImmediate` is not supported in WebView, we should use cross-platform api

File: src/features/MoveItemsFeature.ts [hunk 3]
```diff
@@ -117,9 +121,10 @@ export class MoveItemsFeature implements Feature {
   };
 
   private moveListElementRightCommand = (editor: MyEditor) => {
-    if (this.ime.isIMEOpened()) {
```
Comment (Fyzu) [inline src/features/MoveItemsFeature.ts:120]: On mobile devices, we use the toolbar for command actions, and then `ime` always opened.

On mobiles, while you edit text, your keyboard is always opened, so `isIMEOpened`, and this command won't work. It's necessary to keep it? or how we can handle it? or we can disable this check on mobile?

**Comment by Fyzu** [comment]
@vslinko take a look at the comments, pls :)

File: src/ObsidianOutlinerPlugin.ts (hunk 1 above)
Comment (vslinko) [reply src/ObsidianOutlinerPlugin.ts:40]: Yes, it's ok. This isn't relevant for current version of Obsidian.

**Comment by vslinko** [comment]
@Fyzu Thank you for your contribution! You have done a good job.

I see some problems with PR:

Some tests were broken. I checked the main branch and the tests work there. It needs to fix the code before release. I think it might be because of setTimeout.

I installed the plugin on iOS and it doesn't work. It shows an error when launching the plugin. Have you tested it on iOS?

**Comment by Fyzu** [comment]
> I installed the plugin on iOS and it doesn't work. It shows an error when launching the plugin. Have you tested it on iOS?

Yeah. It works great on my iPhone. But I'll take another check

**Pull request merged**